### PR TITLE
Use fixed size buffers for EC coordinates

### PR DESCRIPTION
**Pull request by csstaub**
Coordinates must have a fixed size.

Changes:
  * Add new constructor for fixed-size byte buffers
  * Add helper functions for curve name/size
  * Update JWK serialization code

R: @dgalling @sqshh

### CODE REVIEW

File: shared.go [hunk 1]
```diff
@@ -182,3 +184,31 @@ func (dst *rawHeader) merge(src *rawHeader) {
...
+	case elliptic.P384():
+		return "P-384", nil
+	case elliptic.P521():
+		return "P-521", nil
+	default:
+		return "", fmt.Errorf("square/go-jose: unsupported/unknown elliptic curve")
+	}
+}
+
+// Get size of curve
```
Comment (dgalling) [inline shared.go]: in bytes?

Comment (csstaub) [inline shared.go]: fixed

File: encoding.go [hunk 2]
```diff
@@ -132,6 +132,14 @@ func newBuffer(data []byte) *byteBuffer {
 	}
 }
 
+func newFixedSizeBuffer(data []byte, length int) *byteBuffer {
```
Comment (alokmenghrajani) [inline encoding.go]: comment saying the buffer is prepended with 0's?

**Comment by alokmenghrajani** [comment]
lgtm

**Comment by csstaub** [comment]
(Sidenote: for reference on the change, see http://www.rfc-editor.org/rfc/rfc7518.txt section 6.2.1.2)

**Pull request merged**